File: src/main/java/org/radarcns/consumer/realtime/action/ActionBase.java
package org.radarcns.consumer.realtime.action;

import java.util.List;
import org.radarcns.config.realtime.ActionConfig;

public abstract class ActionBase implements Action {

  private final List<String> projects;

  public ActionBase(ActionConfig config) {
    this.projects = config.getProjects();
  }

  @Override
  public List<String> getProjects() {
    return projects;
  }
}

File: src/main/java/org/radarcns/consumer/realtime/action/ActionBase.kt
package org.radarcns.consumer.realtime.action

import org.radarcns.config.realtime.ActionConfig

abstract class ActionBase(
        val config: ActionConfig,
        override val projects: List<String>? = config.projects,
        override val subjects: List<String>? = config.subjects,
        override val subjectIdField: String? = config.projectIdField,
        override val projectIdField: String? = config.subjectIdField,
        override val timeField: String? = config.timeField,
        override val sourceIdField: String? = config.sourceIdField,
) : Action